pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -705,208 +704,172 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
+			// We have a non catch all listener on some user specified port
+			// The user specified port may or may not match a service port.
+			// If it does not match any service port and the service has only
+			// one port, then we pick a default service port. If service has
+			// multiple ports, we expect the user to provide a virtualService
+			// that will route to a proper Service.
 
+			listenPort := &model.Port{
+				Port:     int(egressListener.IstioListener.Port.Number),
+				Protocol: model.ParseProtocol(egressListener.IstioListener.Port.Protocol),
+				Name:     egressListener.IstioListener.Port.Name,
+			}
+
+			// If capture mode is NONE i.e., bindToPort is true, and
+			// Bind IP + Port is specified, we will bind to the specified IP and Port.
+			// This specified IP is ideally expected to be a loopback IP.
+			//
+			// If capture mode is NONE i.e., bindToPort is true, and
+			// only Port is specified, we will bind to the default loopback IP
+			// 127.0.0.1 and the specified Port.
+			//
+			// If capture mode is NONE, i.e., bindToPort is true, and
+			// only Bind IP is specified, we will bind to the specified IP
+			// for each port as defined in the service registry.
+			//
+			// If captureMode is not NONE, i.e., bindToPort is false, then
+			// we will bind to user specified IP (if any) or to the VIPs of services in
+			// this egress listener.
+			bind := egressListener.IstioListener.Bind
+			if bindToPort && bind == "" {
+				bind = actualLocalHostAddress
+			} else if len(bind) == 0 {
+				bind = actualWildcard
+			}
+
+			for _, service := range services {
 				listenerOpts := buildListenerOpts{
 					env:            env,
 					proxy:          node,
 					proxyInstances: proxyInstances,
 					proxyLabels:    proxyLabels,
-					port:           servicePort.Port,
-					bind:           "",
-					bindToPort:     false,
+					bind:           bind,
+					port:           listenPort.Port,
+					bindToPort:     bindToPort,
 				}
 
 				pluginParams := &plugin.InputParams{
-					ListenerProtocol:           plugin.ModelProtocolToListenerProtocol(servicePort.Protocol),
+					ListenerProtocol:           plugin.ModelProtocolToListenerProtocol(listenPort.Protocol),
 					DeprecatedListenerCategory: networking.EnvoyFilter_DeprecatedListenerMatch_SIDECAR_OUTBOUND,
 					Env:                        env,
 					Node:                       node,
 					ProxyInstances:             proxyInstances,
 					Push:                       push,
-					Bind:                       "",
-					Port:                       servicePort,
+					Bind:                       bind,
+					Port:                       listenPort,
 					Service:                    service,
 				}
 
-				configgen.buildSidecarOutboundListenerForPortOrUDS(listenerOpts, pluginParams,
-					listenerMap, virtualServices, actualWildcard)
+				configgen.buildSidecarOutboundListenerForPortOrUDS(listenerOpts, pluginParams, listenerMap,
+					virtualServices, actualWildcard)
 			}
-		}
```

Comment:
Can you explain a bit more why this change was made ? Is it related to 'having a default sidecar', or just a separate change ?

Reply:
This change is a weird diff from being unindented. Before we had a special case of sidecarScope == nil, but now we have a default scope for sidecar proxies and gateways have a sidecar scope, so we no longer need the special case